mathjax3-ts/a11y/explorer.ts

```diff
@@ -0,0 +1,263 @@
+
+}
+
+/**
+ * The mixin for adding the Explorer to MathDocuments
+ *
+ * @param {B} BaseMathDocument      The MathDocument class to be extended
+ * @returns {ExplorerMathDocument}  The extended MathDocument class
+ */
+export function ExplorerMathDocumentMixin<B extends MathDocumentConstructor<HTMLDOCUMENT>>(
+    BaseDocument: B
+): MathDocumentConstructor<ExplorerMathDocument> & B {
+
+    return class extends BaseDocument {
+
+        public static OPTIONS: OptionList = {
+            ...BaseDocument.OPTIONS,
+            renderActions: expandable({
+                ...BaseDocument.OPTIONS.renderActions,
+                explorable: [STATE.EXPLORER]
+            })
+        };
+
+        /**
+         * The objects needed for the explorer
+         */
+        public explorerObjects: ExplorerObjects;
+
+        /**
+         * Extend the MathItem class used for this MathDocument
+         *   and create the visitor and exporer objects needed for the explorer
+         *
+         * @override
+         * @constructor
+         */
+        constructor(...args: any[]) {
+            super(...args);
+            const ProcessBits = (this.constructor as typeof BaseDocument).ProcessBits;
+            if (!ProcessBits.has('explorer')) {
+                ProcessBits.allocate('explorer');
+            }
+            const visitor = new SerializedMmlVisitor(this.mmlFactory);
+            const toMathML = ((node: MmlNode) => visitor.visitTree(node));
+            this.options.MathItem = ExplorerMathItemMixin(this.options.MathItem, toMathML);
+            this.explorerObjects = {
+                region: new LiveRegion(this),
+                tooltip: new ToolTip(this),
+                tooltip2: new ToolTip(this),
+                tooltip3: new ToolTip(this),
+                magnifier: new HoverRegion(this)
+            };
+        }
+
+        /**
+         * Add the Explorer to the MathItems in this MathDocument
+         */
+        public explorable() {
+            if (!this.processed.isSet('explorer')) {
+                for (const math of this.math) {
+                    (math as ExplorerMathItem).explorable(this);
+                }
+                this.processed.set('explorer');
+            }
+            return this;
+        }
+
+        /**
+         * @override
+         */
+        public state(state: number, restore: boolean = false) {
+            super.state(state, restore);
+            if (state < STATE.EXPLORER) {
+                this.processed.clear('explorer');
+            }
+            return this;
+        }
+
+    };
+
+}
+
+/*==========================================================================*/
+
+/**
+ * Add Explorer functions to a Handler instance
+ *
+ * @param {Handler} handler   The Handler instance to enhance
+ * @returns {Handler}         The handler that was modified (for purposes of chainging extensions)
+ */
+export function ExplorerHandler(handler: HANDLER, MmlJax: MATHML = null) {
```

Comment:
Because SRE returns a serialized MathML string, that has to be parsed into MathJax's internal representation by the `semantic-enrich` extension.  That requires a MathML input jax, and so `MmlJax` is the one to use.  It *may* be the input jax, but if you are using TeX input, you won't have a MathML input jax for that, so this would be a separate MathML input jax just for the semantic enrichment.